ReactAndroid/src/main/java/com/facebook/react/uimanager/UIViewOperationQueue.java

```diff
@@ -491,6 +497,33 @@ public void execute() {
     }
   }
 
+  private final class TakeSnapshot extends ViewOperation {
+    private Snapshot snapshot;
+    private Promise promise;
+    private File destFile;
+
+    public TakeSnapshot(int tag, Snapshot snapshot, File destFile, Promise promise) {
+      super(tag);
+      this.snapshot = snapshot;
+      this.promise = promise;
+      this.destFile = destFile;
+    }
+    @Override
+    public void execute() {
+      try {
+        View view = mNativeViewHierarchyManager.resolveView(mTag);
+        FileOutputStream fileOutputStream = new FileOutputStream(destFile);
+        snapshot.captureViewToFileOutputStream(view, fileOutputStream);
+        fileOutputStream.close();
```

Comment:
This needs to be in a `finally` section, check out some tutorials, e.g. http://www.quepublishing.com/articles/article.aspx?p=26067